File: sources/common/src/main/java/fr/unice/polytech/si5/al/e/model/Validator.java
package fr.unice.polytech.si5.al.e.model;

import javax.persistence.Entity;
import javax.validation.constraints.NotNull;

@Entity
public class Validator {

    @NotNull
    private boolean insuranceValidation;

    @NotNull
    private boolean pathValidation;


    public Validator(){
        this.insuranceValidation = false;
        this.pathValidation = false;
    }

    public boolean isValid(){
        return insuranceValidation && pathValidation;
    }

    public void setInsuranceValidation(boolean insuranceValidation) {
        this.insuranceValidation = insuranceValidation;
    }

    public void setPathValidation(boolean pathValidation) {
        this.pathValidation = pathValidation;
    }
}

File: blablamoveService/typescript/main/src/entity/Validator.ts
export class Validator {
    private insuranceValidation: boolean;
    private pathValidation: boolean;

    public set $insuranceValidation(value: boolean) {
        this.insuranceValidation = value;
    }

    public set $pathValidation(value: boolean) {
        this.pathValidation = value;
    }

    constructor() {
        this.insuranceValidation = false;
        this.pathValidation = false;
    }

    public isValid(): boolean {
        return this.insuranceValidation && this.pathValidation;
    }
}
